pkg/nats/server.go

```diff
@@ -0,0 +1,74 @@
+package nats
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/bacalhau-project/bacalhau/pkg/model"
+	"github.com/nats-io/nats-server/v2/server"
+	"github.com/rs/zerolog/log"
+)
+
+const ReadyForConnectionsTimeout = 5 * time.Second
+
+type ServerManagerParams struct {
+	Options *server.Options
+}
+
+// ServerManager is a helper struct to manage a NATS server
+type ServerManager struct {
+	Server *server.Server
+}
+
+// NewServerManager is a helper function to create a NATS server with a given options
+func NewServerManager(ctx context.Context, opts *server.Options) (*ServerManager, error) {
+	ns, err := server.NewServer(opts)
+	if err != nil {
+		return nil, err
+	}
+	ns.SetLoggerV2(NewZeroLogger(log.Logger), opts.Debug, opts.Trace, opts.TraceVerbose)
+	go ns.Start()
```

Comment:
creating the server should happen separately from starting it, let's make this a method on `ServerManager`.

Reply:
yeah I understand the benefits of decoupling the creation and starting of the server, specially when we need to register handlers or work with the server before it is started, such as the case with our http server. At the same time it adds some complexity where users need to remember to always start the server, and we need to handle multiple starts or starting a stopped server.

In this specific case, the server needs to be started immediately as a client is created in the following lines in `NATSTransport` and will fail if the server is not started
https://github.com/bacalhau-project/bacalhau/blob/nats/pkg/nats/transport/nats.go#L115